src/js/settings.js

```diff
@@ -506,7 +506,61 @@ function importSubscriptions(){
     })
   });
 }
+/**
+* Import NewPipe Channels Subscriptions
+* @return {Void}
+*/
+function importNewpipeSubscriptions(){
+  // Open user's file browser.  Only show .json files.
+ dialog.showOpenDialog({
+  properties: ['openFile'],
+  filters: [
+    {name: 'JSON', extensions: ['*']},
+  ]
+}, function(fileLocation){
+  if(typeof(fileLocation) === 'undefined'){
+    console.log('Import Aborted');
+    return;
+  }
+  console.log(fileLocation);
+  let i = fileLocation[0].lastIndexOf('.');
+  let fileType = (i < 0) ? '' : fileLocation[0].substr(i);
+  console.log(fileType);
+
+  fs.readFile(fileLocation[0], function(readErr, data){
+    if(readErr){
+      showToast('Unable to read file.  File may be corrupt or have invalid permissions.');
+      throw readErr;
+    }
 
+    if (data.includes("app_version")){
+      let newpipe, n, link, newpipesub;
+      
+      newpipe = JSON.parse(data);
+      
+      for (n in newpipe.subscriptions) {
```

Comment:
This loop will likely take a long time.  Please use the loading bar to visualize the progress.  Take another look at the [OPML Import](https://github.com/FreeTubeApp/FreeTube/blob/3d29ce55a50ae1a3c3093338616e48859a86c61c/src/js/settings.js#L440) function for an example on how to do this.